gateway/mw_graphql_test.go

```diff
@@ -192,113 +192,138 @@ func TestGraphQLMiddleware_EngineMode(t *testing.T) {
 	})
 
 	t.Run("graphql engine v2", func(t *testing.T) {
-		BuildAndLoadAPI(func(spec *APISpec) {
-			spec.UseKeylessAccess = true
-			spec.Proxy.ListenPath = "/"
-			spec.GraphQL.Enabled = true
-			spec.GraphQL.ExecutionMode = apidef.GraphQLExecutionModeExecutionEngine
-			spec.GraphQL.Version = apidef.GraphQLConfigVersion2
-		})
+		t.Run("subgraph", func(t *testing.T) {
+			BuildAndLoadAPI(func(spec *APISpec) {
+				spec.UseKeylessAccess = true
+				spec.Proxy.ListenPath = "/"
+				spec.GraphQL.Enabled = true
+				spec.GraphQL.ExecutionMode = apidef.GraphQLExecutionModeSubgraph
+				spec.GraphQL.Schema = gqlSubgraphSchemaAccounts
+			})
 
-		t.Run("on disabled websockets", func(t *testing.T) {
-			cfg := config.Global()
-			cfg.HttpServerOptions.EnableWebSockets = false
-			config.SetGlobal(cfg)
+			t.Run("should execute subgraph successfully", func(t *testing.T) {
+				request := gql.Request{
+					Query:     gqlSubgraphQueryAccounts,
+					Variables: []byte(gqlSubgraphVariables),
+				}
 
-			t.Run("should respond with 422 when trying to upgrade to websockets", func(t *testing.T) {
-				_, _ = g.Run(t, []test.TestCase{
-					{
-						Headers: map[string]string{
-							headers.Connection:           "upgrade",
-							headers.Upgrade:              "websocket",
-							headers.SecWebSocketProtocol: "graphql-ws",
-							headers.SecWebSocketVersion:  "13",
-							headers.SecWebSocketKey:      "123abc",
-						},
-						Code:      http.StatusUnprocessableEntity,
-						BodyMatch: "websockets are not allowed",
-					},
-				}...)
+				_, _ = g.Run(t, test.TestCase{
+					Data:      request,
+					BodyMatch: `variables.*_representations.*__typename.*User.*id.*1.*query Subgraph\(\$_representations: \[_Any!\]!\) {.*_entities\(representations: \$_representations\) {.*on User {.* id.* username.*}.*}.*}.*}`,
```

Comment:
does it make sense to use body match func here instead of regexp to make clear what we are asserting?

Reply:
Do you have an example?